host/pygreat/comms_backends/usb.py

```diff
@@ -98,16 +95,103 @@ def __init__(self, **device_identifiers):
             raise DeviceNotFoundError()
 
         # For now, supported boards provide a single configuration, so we
-        # can accept the first configuration provided.
-        self.device.set_configuration()
+        # can accept the first configuration provided. If the device isn't
+        # already configured, apply that configuration.
+        #
+        # Note that we can't universally apply configurations, as e.g. linux
+        # doesn't support this, and macOS considers setting the device's configuration
+        # grabbing an exclusive hold on the device. Both set the configuration for us,
+        # so this is skipped.
+        if not self.device.get_active_configuration():
+            self.device.set_configuration()
 
         # Start off with no knowledge of the device's state.
         self._last_command_arguments = None
+        self._have_exclusive_access = False
 
         # Run the parent initialization.
         super(USBCommsBackend, self).__init__(**device_identifiers)
 
 
+    def _hold_libgreat_interface(self, timeout=1000):
+        """
+        Ensures we have exclusive access to the USB interface used by libgreat.
+
+        This is used internally both to get long-term exclusive access and to
+        take temporary exclusive access (e.g. during a libgreat command).
+
+        parameters:
+            timeout -- the maximum amount of time we should wait before
+        """
+
+        # If we already have exclusive access, there's no need to grab it again.
+        if self._have_exclusive_access:
+            return
+
+        # If timeout is none, set it to a microsecond.
+        if timeout is None:
+            timeout = 0.001
+
+        # Claim the first interface on the device, which we consider the standard
+        # interface used by libgreat.
+        timeout = time.time() + (timeout / 1000)
+
+        while True:
+            try:
+                usb.util.claim_interface(self.device, 0)
+                return
+            except usb.core.USBError as e:
+
+                # If we have EBUSY (linux) or EACCES (macos), or None (windows), try again.
+                if e.errno in (errno.EBUSY, errno.EACCES, None):
+                    pass
+                else:
+                    raise
+
+            if time.time() > timeout:
+                raise IOError("timed out trying to claim access to a GreatFET device!")
```

Comment:
>`timed out trying to claim access to a GreatFET device!`

Should probably change this to refer to libgreat devices in general, since libgreat isn't specific to GreatFET.